test/setup/make_source.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+echo "#!/usr/bin/env bash" > ../source.sh
+
+project_id=$(terraform output project_id)
+echo "export TF_VAR_project_id='$project_id'" >> ../source.sh
+
+sa_json=$(terraform output sa_key)
+# shellcheck disable=SC2086
+# shellcheck disable=SC2129
+echo "export SERVICE_ACCOUNT_JSON='$(echo $sa_json | base64 --decode)'" >> ../source.sh
+
+echo "export TF_VAR_region=\"us-east4\"" >> ../source.sh
+echo "export TF_VAR_zones=\"[\\\"us-east4-a\\\",\\\"us-east4-b\\\",\\\"us-east4-c\\\"]\"" >> ../source.sh
+
+compute_engine_service_account=$(terraform output compute_engine_service_account)
```

Comment:
This is not correct, please try to understand the flow.

The account created by `prepare` is used to EXECUTE the Terraform flow. It must NOT be used within the fixtures themselves.

If the fixtures themselves need a service account, create it inside them or create a separate account.

To explain further, there is absolutely no reason to run GKE nodes with the full permissions that Terraform has.

If you absolutely need to create a service account for usage IN the fixtures, this should be a seperate account which is created with its own permissions/roles.

Reply:
Added creation of separated SA